File: QuickMobileTest/QuickMobileTest/Data/Resource List/Entity/ResourceResponse.swift
//
//  ResourceResponse.swift
//  QuickMobileTest
//
//  Created by Balaji Galave on 5/18/19.
//

import Foundation

struct ResourceResponse : Codable {
    let _id: String = ""
    let slug: String?
    let eid: String?
    let title: String?
    let description: String?
    let category_eid: String?
    let __v: Int = 0
    let photo: String?
    let _active: Bool = false
    let updated_at: String?
    let created_at: String?
    let addresses: [AddressResponse]?
    let contactInfo: ContactInfoResponse?
    let bizHours: BizHourResponse?
    let socialMedia: SocialMediaResponse?
}

struct ContactInfoResponse : Codable {
    let website: [String]?
    let email: [String]?
    let phoneNumber: [String]?
    let tollFree: [String]?
    let faxNumber: [String]?
}

struct BizHourResponse: Codable {
    let sunday: Day?
    let monday: Day?
    let tuesday: Day?
    let wednesday: Day?
    let thursday: Day?
    let friday: Day?
    let saturday: Day?
}

struct Day: Codable {
    let from: String?
    let to: String?
}

struct SocialMediaResponse: Codable {
    let youtubeChannel: [String]?
    let twitter: [String]?
    let facebook: [String]?
}

File: app/src/main/java/com/quickseries/mobiletest/data/resources/ResourceResponse.kt
package com.quickseries.mobiletest.data.resources

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.util.*

/**
 * Response item containing informations about a resource.
 */
@JsonClass(generateAdapter = true)
data class ResourceResponse(
    @Json(name = "_id")
    val id: String? = null,
    @Json(name = "created_at")
    val createdAt: Date? = null,
    @Json(name = "updated_at")
    val updatedAt: Date? = null,
    @Json(name = "slug")
    val slug: String? = null,
    @Json(name = "category_eid")
    val categoryEid: String? = null,
    @Json(name = "eid")
    val eid: String? = null,
    @Json(name = "title")
    val title: String? = null,
    @Json(name = "description")
    val description: String? = null,
    @Json(name = "__v")
    val v: Int? = null,
    @Json(name = "_active")
    val active: Boolean = false,
    @Json(name = "photo")
    val photoUrl: String? = null,
    @Json(name = "contactInfo")
    val contactInfo: ContactInfoResponse = ContactInfoResponse(),
    @Json(name = "addresses")
    val addresses: List<AddressResponse> = emptyList(),
    @Json(name = "bizHours")
    val bizHours: BizHoursResponse = BizHoursResponse(),
    @Json(name = "socialMedia")
    val socialMedia: SocialMediaResponse = SocialMediaResponse()
)